### DISCUSSION TITLE
App Intents for Shortcuts app integration

### DISCUSSION THREAD

**Discussion by sagemintblue** [MonitorControl/MonitorControl#1519] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Is your feature request related to a problem? Please describe

I'd like to be able to use the Shortcuts app [1] to trigger MonitorControl actions e.g. set a particular screen brightness to X%. See App Intents documentation [2] for further background.

[1] https://support.apple.com/guide/shortcuts-mac/welcome/mac
[2] https://developer.apple.com/documentation/appintents/app-shortcuts

### Describe the solution you'd like

I'd like to see MonitorControl actions exposed via the App Intents framework. It may make sense to start with a single intent that allows a particular screen brightness to be set directly to a target percentage.

### Describe alternatives you've considered

https://github.com/kfix/ddcctl for command line monitor brightness control along with the existing shell command execution support in the Shortcuts app.

I'd rather have a single monitor control app and avoid additional dependencies.

**Comment by waydabber** [comment]
Hi there, this probably won't be added to MonitorControl as development generally ceased, however [BetterDisplay](https://betterdisplay.pro) now supports various integration methods (CLI, URL schema, HTTP etc) and that can be easily used with Shortcuts and App Intents support is right around the corner as well (implementation is done, just need to populate with some useful actions).

https://github.com/waydabber/BetterDisplay/wiki/Integration-features,-CLI
https://github.com/waydabber/BetterDisplay/issues/1395
